vizro-core/src/vizro/models/_components/table.py

```diff
@@ -12,14 +12,26 @@
 import vizro.tables as vt
 from vizro.managers import data_manager
 from vizro.models import Action, VizroBaseModel
-from vizro.models._action._actions_chain import _action_validator_factory
+from vizro.models._action._actions_chain import _set_actions
 from vizro.models._components._components_utils import _process_callable_data_frame
 from vizro.models._models_utils import _log_call
 from vizro.models.types import CapturedCallable
 
 logger = logging.getLogger(__name__)
 
 
+def _get_table_type(figure):  # this function can be applied also in pre-build
+    kwargs = figure._arguments.copy()
+
+    # This workaround is needed because the underlying table object requires a data_frame
+    kwargs["data_frame"] = DataFrame()
+
+    # The underlying table object is pre-built, so we can fetch its ID.
+    underlying_table_object = figure._function(**kwargs)
```

Comment:
Posting the answer @petar-qb gave to me:

Graphs and Tables in the Dash (so in the Vizro too) are handled differently.

dcc.Graph has: `id`, `figure` attributes where `figure` attribute is any plotly chart.
If this graph has to be changed as Output of the callback we target it as - dash.Output(`dcc_graph_id`, `"figure"`)
If this graph has to be an Input (let's say `clickData` property) of the callback we defined it as - dash.Input(`dcc_graph_id`, `"clickData"`)
It means that we access to graph properties by accessing the outer wrapper dcc.Graph component.

The problem is that there's no outer dcc component wrapper for tables :confused:. So, there is nothing like dcc.Table Dash inbuilt component that has `id` and `figure` attributes. Table ID is written `directly` inside its "figure" callable (e.g. inside the `dash_table.DataTable()`). This is different than graphs because plotly graph doesn't contain the ID (you cannot put the ID inside px.box(...)), but its outer component dcc.Graph does.

Yes, Vizro has created some kind of wrapper `vm.Table` that has `id` and `figure` where the figure is callable that can return dash_table.DataTable or AgGrid. Still, we didn't solve this problem because we can't (or at least, we didn't decide like that) to propagate the `vm.Table.id` into underlying table component.

Now, let's give an example on how callback inputs and outputs are created in the case of Tables.
If Vizro Table has to be changed as Output of the callback we target it with - dash.Output(`vm_table_id`, `"children"`) - We can re-render dash_table.DashTable only if we change "children" of the outer Div compoennt.
If Vizro Table has to be the Input (let's say `active_cell` property) of the callback we defined it as - dash.Input(`underlying_table_id`, `"figure"`) - So we need to fetch this ID in the case that filter_interaction is defined on the Table.

Reply:
@petar-qb this half makes sense to me. The half that doesn't make sense is:
1. why can't we also use `self._callable_object_id` in callback outputs as well? Is the same true for AG Grid or just Dash datatable?
2. why not set `self._callable_object_id = self.id` - you say we decided not to? I guess the answer is that if my above question is not possible you'll get duplicate ids for the table and the containing div
3. do we keep on setting `_callable_object_id` somewhere outside `Table.build`?

Ideally what I'd like to do is this:
```
class Table(VizroBaseModel):
    def __call__(self, **kwargs):
        kwargs.setdefault("data_frame", data_manager._get_component_data(self.id))
        return self.figure(id=self.id, **kwargs)

    # no need for pre_build at all
```
so that the `id` is always injected from the automatically. But I'm guessing this will not be possible.

No need resolve this conversation as part of this PR because it's outside the scope of the PR, but it would be great to have a chat about it - let me put something in the calendar 🙂